src/Arcus.Observability.Telemetry.OpenTelemetry/OpenTelemetryObservability.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Linq;
+using OpenTelemetry;
+
+namespace Arcus.Observability.Telemetry.OpenTelemetry
+{
+    /// <summary>
+    /// Represents the OpenTelemetry implementation of the <see cref="IObservability"/> interface
+    /// to track telemetry throughout the application in a developer-friendly way.
+    /// </summary>
+    internal class OpenTelemetryObservability : IObservability
+    {
+        private readonly ActivitySource _applicationSource;
+        private readonly IMeterFactory _meterFactory;
+        private readonly OpenTelemetryObservabilityOptions _options;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenTelemetryObservability"/> class.
+        /// </summary>
+        internal OpenTelemetryObservability(ActivitySource applicationSource, IMeterFactory meterFactory, OpenTelemetryObservabilityOptions options)
+        {
+            ArgumentNullException.ThrowIfNull(applicationSource);
+            ArgumentNullException.ThrowIfNull(meterFactory);
+            ArgumentNullException.ThrowIfNull(options);
+
+            _applicationSource = applicationSource;
+            _meterFactory = meterFactory;
+            _options = options;
+        }
+
+        /// <summary>
+        /// Tracks a custom request with contextual information to the configured observability backend.
+        /// </summary>
+        /// <remarks>
+        ///     Make sure to dispose the returned operation result to flush the telemetry to the observability backend.
+        /// </remarks>
+        /// <param name="operationName">The name of the request operation.</param>
+        /// <param name="telemetryContext">The additional contextual information to provide context to the request.</param>
+        /// <returns>
+        ///     The scoped request operation that allows to specify whether the request operation was successful.
+        /// </returns>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="operationName"/> is blank.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="telemetryContext"/> is <c>null</c>.</exception>
+        public RequestOperationResult StartCustomRequest(string operationName, IDictionary<string, object> telemetryContext)
+        {
+            ArgumentException.ThrowIfNullOrWhiteSpace(operationName);
+            ArgumentNullException.ThrowIfNull(telemetryContext);
+
+            using var activity = _applicationSource.StartActivity(operationName, ActivityKind.Server);
```

Comment:
Should the `ActivtyKind` be something that the caller can specify/have control over ?
Of course, this would mean that we need to introduce our own enum-type that we then map to the `ActivityKind` enum type here.

Reply:
I would say not, as any situation that would require that low level manipulation is also not really tracking requests anymore. It follows Microsoft's docs, but nothing stops people to do activity tracing with the registered activity source themselves.